journeyapps-db/src/types/QueryType.ts

```diff
@@ -0,0 +1,34 @@
+import { QueryType as SchemaQueryType, QueryTypeFactory as SchemaQueryTypeFactory } from '@journeyapps/parser-schema';
+import { Query } from '../query/Query';
+import { ObjectType } from './ObjectType';
+import { DBTypeMixin } from './Type';
+
+export class QueryType extends DBTypeMixin(SchemaQueryType) {
+  clone = function (query: Query) {
+    return query._clone();
+  };
+  cast = function (this: QueryType, value: any) {
+    if (typeof value != 'object') {
+      throw new Error(value + ' is not an object');
+    }
+    const thisTypeName = this.objectType.name;
+    if (
+      value.type != null &&
+      value.type instanceof ObjectType &&
```

Comment:
possible copy paste of this entire function? should probably be `Query` and  as well as above on like 7